build-system/babel-plugins/babel-plugin-amd-custom-define/index.js

```diff
@@ -0,0 +1,61 @@
+const {readFileSync} = require('fs');
+const {join: pathJoin, relative} = require('path');
+
+const pathToModuleName = (filename) =>
+  filename.replace(/^(\.\/)?dist\//, '').replace(/(\.max)?\.m?js$/, '');
+
+let defineTemplate;
+
+module.exports = function (babel) {
+  const {types: t} = babel;
+
+  /** @return {string} */
+  function getTemplate() {
+    if (!defineTemplate) {
+      defineTemplate = readFileSync(
+        pathJoin(__dirname, 'define-template.js'),
+        'utf8'
+      );
+    }
+    return defineTemplate;
+  }
+
+  return {
+    name: 'amd-custom-define',
+    visitor: {
+      CallExpression(path, state) {
+        const callee = path.get('callee');
+        if (!callee.isIdentifier({name: 'define'})) {
```

Comment:
I think our dependency on the `modules-amd` transform is just so that we can extract this step? If so, we can actually just process the source text and extract out import declarations, no?

Reply:
True. It seems like it's not too difficult using `@babel/helper-module-transforms`, so I'm doing that now.